pkg/asset/quota/openstack/openstack.go

```diff
@@ -47,6 +87,7 @@ func compute(ci *validation.CloudInfo, replicas []int64, machines []*openstackpr
 		ramConstraint := machineFlavorRAMToQuota(flavorInfo, ci)
 		ramConstraint.Count = ramConstraint.Count * replicas[idx]
 		ret = append(ret, instanceConstraint(int64(replicas[idx])), coresConstraint, ramConstraint)
+		ret = append(ret, generateConstraint("Port", int64(len(m.Networks))))
```

Comment:
On line 89you might be setting N times a requirement for the resources, instead of adding requirements from multiple MachineSets

Reply:
I think it's slightly clearer in my refactor #4483 how these are added up. These are run through aggregate() below before being returned.

Reply:
> On line 89 you might be setting N times a requirement for the resources, instead of adding up requirements from multiple MachineSets

nice catch, thanks.

Reply:
> I think it's slightly clearer in my refactor #4483 how these are added up. These are run through aggregate() below before being returned.

ack, happy to rebase if yours merges before mine!